nodejs/eks/nodegroup.ts

```diff
@@ -1093,19 +1093,19 @@ function createNodeGroupV2Internal(
         return args.instanceProfile?.arn ?? c.nodeGroupOptions.instanceProfile!.arn;
     });
 
-    core.apply((c) => {
-        if (c.nodeGroupOptions.nodeSecurityGroup && args.nodeSecurityGroup) {
-            if (
-                c.nodeSecurityGroupTags &&
-                c.nodeGroupOptions.nodeSecurityGroup.id !== args.nodeSecurityGroup.id
-            ) {
-                throw new pulumi.ResourceError(
-                    `The NodeGroup's nodeSecurityGroup and the cluster option nodeSecurityGroupTags are mutually exclusive. Choose a single approach`,
-                    parent,
-                );
+    const coreSecurityGroupId = core.nodeGroupOptions.nodeSecurityGroup?.apply((sg) => sg?.id);
+    pulumi
+        .all([coreSecurityGroupId, args.nodeSecurityGroup?.id, core.nodeSecurityGroupTags])
+        .apply(([coreSecurityGroup, nodeSecurityGroup, sgTags]) => {
+            if (coreSecurityGroup && nodeSecurityGroup) {
+                if (sgTags && coreSecurityGroup !== nodeSecurityGroup) {
```

Comment:
Always suspicious of equality but this look like it's over string right? Might be OK. Perhaps call the variables "coreSecurityGroupId" or some such.

Reply:
Yeah it's over string. Noticed that it was comparing `Output<string>` before..